_posts/2023-03-29-kubeflow-1.7-release.md

```diff
@@ -0,0 +1,190 @@
+
+<br />
+
+* Narrow down Katib RBAC rules ([#2091](https://github.com/kubeflow/katib/pull/2091))
+* Support Postgres as a Katib DB ([#1921](https://github.com/kubeflow/katib/pull/1921))
+* [SDK] Create Tune API in the Katib SDK ([#1951](https://github.com/kubeflow/katib/pull/1951))
+* [SDK] Get Trial Metrics from Katib DB ([#2050](https://github.com/kubeflow/katib/pull/2050))
+* More Suggestion container fields in Katib Config ([#2000](https://github.com/kubeflow/katib/pull/2000))
+* Katib UI: Enable pagination/sorting/filtering ([#2017](https://github.com/kubeflow/katib/pull/2017) and [#2040](https://github.com/kubeflow/katib/pull/2040))
+* Katib UI: Add authorization mechanisms ([#1983](https://github.com/kubeflow/katib/pull/1983))
+
+<br />
+
+### Training operator
+
+Kubeflow’s unified distributed training operator enhancements include configuration options for fine tuned resource scaling (processor, memory, storage). It now includes HPA support for Pytorch Elastic workloads where users can specify target metric/utilization in Job Spec. This is used for automatic scale up/down of Pytorch Job matching demand while ensuring the elastic policy configured by the user.  This simplifies user workflows significantly and reduces operational toil and costs. It supports multiple scheduler types: Kubernetes, volcano, custom.   Major 1.7 training operator features include:
+
+<br />
+
+* PodGroup enhancements([#1574](https://github.com/kubeflow/training-operator/issues/1574))
+*  Integration with other training frameworks - Paddlepaddle([#520](https://github.com/kubeflow/community/issues/520))
+*  Enhancements on Pytorch Elastic training ([#1645](https://github.com/kubeflow/training-operator/issues/1645), [#1626](https://github.com/kubeflow/training-operator/issues/1626))
+*  Support coscheduling plugin ([#1722](https://github.com/kubeflow/training-operator/issues/1722))
+*  [SDK] Create Unify Training Client([#1719](https://github.com/kubeflow/training-operator/pull/1719))
+
+<br />
+
+### Pipelines
+
+Model developers recognize the time-saving pythonic workflows in Kubeflow Pipelines, which speed iteration by not requiring the generation of new images for pre-prod experimentation.   The new V2 UI and SDK in Kubeflow 1.7 provide valuable details on each pipeline step.   This simplifies the correlation and analysis of parameters, metadata and artifacts during iteration.
+
+In 1.7, KFP also includes enhancements for [control flow functionality](https://www.kubeflow.org/docs/components/pipelines/v2/author-a-pipeline/pipelines/#dsl-control-flow-features).  There are multiple scenarios when a model developer needs to use control flow and pipeline topology enhancements to define a sub-graph in a pipeline ([#8326](https://github.com/kubeflow/pipelines/pull/8326)):
+
+<br />
+
+* ParallelFor ( [#8146](https://github.com/kubeflow/pipelines/pull/8146)) : 
```

Comment:
This is a SDK only change that currently not supported by the backend. We should remove this.

Reply:
@chensun thanks for this update.   I am a bit confused, but I guess that might be easy with the backend/SDK have different feature sets...I thought the backend was a superset of the SDK.  

That aside, what is the purpose of the Parallelfor feature in the SDK ?   When will the backend be enabled to support Parallelfor ?

Reply:
KFP SDK serves not only KFP backend but whoever supports/implements the IR pipeline spec. So there might be different features sets in the short term. In the long term we hope KFP backend would support all platform-agnostic features plus some KFP-specific features. :) 

For the `Parallelfor` feature itself, the main functionality has been supported by both KFP backend and KFP SDK for a long time. The PR referenced above is to allow setting a `parallelism` limit for `ParallelFor`. Without this setting, `ParallelFor` still works, and it runs all jobs in parallel as long as the cluster resource allows.